### PR TITLE
AW - Driver Availability Controller

### LINKED ISSUE #11: EPIC: Driver Availability: #2 DriverAvailabilityController

### PR DESCRIPTION
**Pull request by ashton-wong**
In this PR, I added the Driver Availability Controller, as well as tests that pass jacoco coverage and pitest mutation testing.

To check this PR, you can run locally and go into swagger UI. Then test that all CRUD operations work.

Closes #11

### CODE REVIEW

**Comment by elaineeeho** [comment]
LGTM

**Review by elaineeeho — DISMISSED** [review]

**Review by michaelw1985 — APPROVED** [review]
LGTM

File: src/main/java/edu/ucsb/cs156/gauchoride/controllers/DriverAvailabilityController.java [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+                .orElseThrow(() -> new EntityNotFoundException(DriverAvailability.class, id));;
+        } else {
+            availability = driverAvailabilityRepository.findByIdAndDriverId(id, getCurrentUser().getUser().getId())
+                .orElseThrow(() -> new EntityNotFoundException(DriverAvailability.class, id));
+        }
+
+        return availability;
+    }
+
+    @Operation(summary = "Create a new ride")
```
Comment (andrewpengucsb) [inline src/main/java/edu/ucsb/cs156/gauchoride/controllers/DriverAvailabilityController.java:68]: Says `Create a new ride` but should be `Create a new driver availability` or similar

File: src/main/java/edu/ucsb/cs156/gauchoride/controllers/DriverAvailabilityController.java [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+        @RequestParam String day,
+
+        @Parameter(name="startTime", description="String, Time the ride starts HH:MM(A/P)M", example="12:30AM", required = true)
+        @RequestParam String startTime,
+
+        @Parameter(name="endTime", description="String, Time the ride ends HH:MM(A/P)M", example="12:30AM", required = true)
+        @RequestParam String endTime,
+
+         @Parameter(name="notes", description="String, Extra information", example="We have two people riding", required = true)
+        @RequestParam String notes
```
Comment (andrewpengucsb) [inline src/main/java/edu/ucsb/cs156/gauchoride/controllers/DriverAvailabilityController.java:83]: Descriptions still talk about rides

File: src/main/java/edu/ucsb/cs156/gauchoride/controllers/DriverAvailabilityController.java [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+        availability.setStartTime(startTime);
+        availability.setEndTime(endTime);
+        availability.setNotes(notes);
+
+        DriverAvailability savedAvailability = driverAvailabilityRepository.save(availability);
+
+        return savedAvailability;
+    }
+
+    @Operation(summary = "Delete an availability if owned by current user")
```
Comment (andrewpengucsb) [inline src/main/java/edu/ucsb/cs156/gauchoride/controllers/DriverAvailabilityController.java:100]: Admin should also be able to delete anyones availability, sorry for being unclear in the issue description

File: src/main/java/edu/ucsb/cs156/gauchoride/controllers/DriverAvailabilityController.java [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+        availability = driverAvailabilityRepository.findByIdAndDriverId(id, getCurrentUser().getUser().getId())
+            .orElseThrow(() -> new EntityNotFoundException(DriverAvailability.class, id));
+        
+
+        driverAvailabilityRepository.delete(availability);
+        return genericMessage("Availability with id %s deleted".formatted(id));
+    }
+
+
+    @Operation(summary = "Edits an availability if owned by current user")
```
Comment (andrewpengucsb) [inline src/main/java/edu/ucsb/cs156/gauchoride/controllers/DriverAvailabilityController.java:119]: Admin should be able to edit, again my fault for not being clear in the issue description

**Review by andrewpengucsb — CHANGES_REQUESTED** [review]

**Review by nguyenliliana — APPROVED** [review]
LGTM

**Review by nguyenliliana — APPROVED** [review]
LGTM

**Review by elaineeeho — APPROVED** [review]
LGTM

**Review by andrewpengucsb — APPROVED** [review]

**Pull request merged**